### PR TITLE
feat(connect-steps): Use step function to config connector 

### PR DESCRIPTION
**Pull request by benjaminpaige**
## Purpose

This change rewrites the configureConnectors lambda function as a step function, replicating intended functionality exactly.

#### Linked Issues to Close

N/A

## Approach

The goal of working this issue was to improve the way in which we configure Kafka Connect connectors.

This is re-application of this [change](https://github.com/CMSgov/cms-bigmac/pull/541) which serves the same purpose in the cms-bigmac project

----- 

below are notes copied from that change 

The goal of working this issue was to improve the way in which we configure Kafka Connect connectors.

The previous design of our configureConnectors appliance (which this PR replaces) went like this:

The connectors are configured by a lambda function called configureConnectors
the lambda function is built with the connector configuration, less env specific variables, supplied from source.
the environment specific material (like usernames, passwords, etc) that a connector configuration might need are resolved from secrets manager at deploy time and set in the lambda as environment variables.
the lambda is trigger immediately following deployment, as a serverless hook.
There were a few issues we noticed with the previous design:

the lambda function was not very tolerant of kafka connect not yet being ready to accept connections. Despite out best efforts in making it tolerant, we would continue to see cases where the lambda did not behave as we expected.
the lambda was really complex. it did many different steps to configure a connector. many different awaits. a lot of looping.
the lambda had environment specific info, often times secret material, set as environment variables in the function. While the lambda execution is secure, this wasn't ideal.
Addressing those issues....
This new pattern goes like this:

A step function is defined to replace the configureConnectors lambda function.
this step function is of the 'EXPRESS' type, which is able to be invoked synchronously and has a 5 minute timeout.
this step function is comprised of many smaller lambda fuctions, along with decision steps, waits, and looping logic.
the step function is designed to configure one connector per invocation, rather than many.
the step function is invoked directly after deployment as a serverless hook, just as the lambda was.
the connector config for each connector is uploaded to Secrets Manager with Cloudformation.
the step function is invoked with a path to a config in secrets manager. during its execution, it fetches the config.
The new design has shown to work well. The much more powerful set of tools step functions provides for orchestrating a complex workflow such as this has proved super helpful. It was really simple to break apart the logic and model it. The step function now waits as intended when the Connect service isn't up yet. It correctly fails the build when the execution fails. The change to how we store connector config keeps potentially sensitive material out of lambdas.

As a side effect, the testConnectors lambda was updated to fetch its connectors list from secrets manager. It was not rewritten as a step function, although I'd recommend making that change when we next have to modify it.

### CODE REVIEW

File: src/libs/connect-lib.js [hunk 1]
```diff
@@ -167,5 +159,89 @@ export async function testConnectors(cluster, service, connectors) {
...
+    (x) => x.name === "privateIPv4Address"
+  )[0].value;
+  console.log(ip);
+  return ip;
+}
+
+export async function checkIfConnectIsReady(ip) {
+  var ready = false;
+  try {
+    const res = await axios.get(`http://${ip}:8083/`);
```
Comment (13bfrancis) [inline src/libs/connect-lib.js:198]: Is the port something that will never change or should this be a variable

File: src/libs/connect-lib.js [hunk 2]
```diff
@@ -167,5 +159,89 @@ export async function testConnectors(cluster, service, connectors) {
       console.log(`Testing connector: ${connector.name}`);
       return testConnector(workerIp, connector);
     })
+  ).then((res) => {
+    return res;
+  });
```
Comment (JonHolman) [inline src/libs/connect-lib.js:164]: Isn't `.then((res) => { return res; }});` the equivalent of not having the `.then` at all?

File: src/libs/connect-lib.js [hunk 3]
```diff
@@ -167,5 +159,89 @@ export async function testConnectors(cluster, service, connectors) {
...
+    })
   );
+  if (taskArns.length === 0) {
+    throw `No task found for cluster ${cluster}`;
+  }
+  const tasks = (
+    await client.send(
+      new DescribeTasksCommand({
+        cluster: cluster,
+        tasks: [taskArns[0]],
```
Comment (JonHolman) [inline src/libs/connect-lib.js:182]: could the `taskArns` array ever have more than one item?

File: src/libs/connect-lib.js [hunk 4]
```diff
@@ -167,5 +159,89 @@ export async function testConnectors(cluster, service, connectors) {
...
+export async function checkIfConnectIsReady(ip) {
+  var ready = false;
+  try {
+    const res = await axios.get(`http://${ip}:8083/`);
+    if (res.status && res.status == 200) {
+      console.log("Kafka Connect service is ready");
+      ready = true;
+    }
+  } catch (error) {
+    console.log(
```
Comment (JonHolman) [inline src/libs/connect-lib.js:204]: should this be `console.error` ?

File: src/libs/connect-lib.js [hunk 5]
```diff
@@ -167,5 +159,89 @@ export async function testConnectors(cluster, service, connectors) {
...
+    console.log(error);
+    console.log("Connector was NOT successfully created.");
+  } finally {
+    return retVal;
+  }
+}
+
+async function fetchConnectorConfigFromSecretsManager(connectorConfigSecret) {
+  console.log(
+    `Fetching connector config from Secrets Manager at:  ${connectorConfigSecret}`
```
Comment (JonHolman) [inline src/libs/connect-lib.js:238]: Could we use SSM parameter store for this?

**Review by 13bfrancis — APPROVED** [review]

File: src/libs/connect-lib.js (hunk 5 above)
Comment (JonHolman) [reply src/libs/connect-lib.js:240]: sounds good.

File: src/libs/connect-lib.js (hunk 3 above)
Comment (JonHolman) [reply src/libs/connect-lib.js:183]: How about changing this:
```
if (taskArns.length === 0) {
    throw `No task found for cluster ${cluster}`;
  }
```
to
```
if (taskArns.length !== 1) {
    throw `taskArns for cluster ${cluster} is not equal to 1`;
  }
```
or that may be too clunky.  I think we should throw an error if its not 1 item.  So we ensure we're appropriately dealing with multiple items when/if that happens.

**Pull request merged**